In [1]:
! conda list | grep pdal

# packages in environment at /home/ec2-user/miniconda3/envs/pdal_jupyter:
pdal                      2.4.0                hec91d99_4    conda-forge
pdal-plugins              1.2.0                    pypi_0    pypi
python-pdal               3.1.2           py310h6245e14_1    conda-forge


In [2]:
! pdal info

PDAL: No input file specified.

usage: pdal info [options] [input]
standard options:
  --developer-debug  Enable developer debug (don't trap exceptions)
  --label            A string to label the process with
  --driver           Override reader driver
  --help             Print help and exit
options:
  --input, -i               Input file name
  --all                     Dump statistics, schema and metadata
  --point, -p               Point to dump --point="1-5,10,100-200" (0 indexed)
  --query                   Return points in order of distance from the
      specified location (2D or 3D) --query Xcoord,Ycoord[,Zcoord][/count]
  --stats                   Dump stats on all points (reads entire dataset)
  --boundary                Compute a hexagonal hull/boundary of dataset
  --dimensions              Dimensions on which to compute statistics
  --enumerate               Dimensions whose values should be enumerated
  --schema                  Dump the schema
  --pipeline-serialization

# Workshop STuff
- https://pdal.io/workshop/exercises/index.html

In [4]:
! aws s3 ls eccoe-lidar/

                           PRE CO_Eastern_ElPaso_2018/
                           PRE ajit/
                           PRE las/
2022-02-25 20:40:59         44 readme.txt


In [5]:
! aws s3 ls eccoe-lidar/CO_Eastern_ElPaso_2018/

2022-03-27 16:37:49   11449736 USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz
2022-03-27 16:37:49   27172583 USGS_LPC_CO_EasternColorado_2018_A18_LD31201353.laz
2022-03-27 16:37:49   16713975 USGS_LPC_CO_EasternColorado_2018_A18_LD31201356.laz
2022-03-27 16:37:49   24014956 USGS_LPC_CO_EasternColorado_2018_A18_LD31201359.laz
2022-03-27 16:37:50   21893078 USGS_LPC_CO_EasternColorado_2018_A18_LD31201362.laz
2022-03-27 16:37:50   16514681 USGS_LPC_CO_EasternColorado_2018_A18_LD31201365.laz
2022-03-27 16:37:50   12999817 USGS_LPC_CO_EasternColorado_2018_A18_LD31201368.laz
2022-03-27 16:37:51    8293389 USGS_LPC_CO_EasternColorado_2018_A18_LD31201371.laz
2022-03-27 16:37:51    1287083 USGS_LPC_CO_EasternColorado_2018_A18_LD31201374.laz
2022-03-27 16:37:51   15866610 USGS_LPC_CO_EasternColorado_2018_A18_LD31231350.laz
2022-03-27 16:37:51   39067512 USGS_LPC_CO_EasternColorado_2018_A18_LD31231353.laz
2022-03-27 16:37:51   38800649 USGS_LPC_CO_EasternColorado_2018_A18_LD31231356.laz
2022

2022-03-27 16:38:40   58426380 USGS_LPC_CO_EasternColorado_2018_A18_LD31441365.laz
2022-03-27 16:38:41   69981986 USGS_LPC_CO_EasternColorado_2018_A18_LD31441368.laz
2022-03-27 16:38:41  119448409 USGS_LPC_CO_EasternColorado_2018_A18_LD31441371.laz
2022-03-27 16:38:42  122624547 USGS_LPC_CO_EasternColorado_2018_A18_LD31441374.laz
2022-03-27 16:38:42  120982466 USGS_LPC_CO_EasternColorado_2018_A18_LD31441377.laz
2022-03-27 16:38:42  124390787 USGS_LPC_CO_EasternColorado_2018_A18_LD31441380.laz
2022-03-27 16:38:43   79616810 USGS_LPC_CO_EasternColorado_2018_A18_LD31441383.laz
2022-03-27 16:38:45   61970586 USGS_LPC_CO_EasternColorado_2018_A18_LD31441386.laz
2022-03-27 16:38:45   63107677 USGS_LPC_CO_EasternColorado_2018_A18_LD31441389.laz
2022-03-27 16:38:45   61750214 USGS_LPC_CO_EasternColorado_2018_A18_LD31441392.laz
2022-03-27 16:38:45   49449937 USGS_LPC_CO_EasternColorado_2018_A18_LD31441395.laz
2022-03-27 16:38:45   51959625 USGS_LPC_CO_EasternColorado_2018_A18_LD31441398.laz
2022

2022-03-27 16:40:01   74925356 USGS_LPC_CO_EasternColorado_2018_A18_LD31621314.laz
2022-03-27 16:40:01   69149497 USGS_LPC_CO_EasternColorado_2018_A18_LD31621317.laz
2022-03-27 16:40:02   61767270 USGS_LPC_CO_EasternColorado_2018_A18_LD31621320.laz
2022-03-27 16:40:03   66853398 USGS_LPC_CO_EasternColorado_2018_A18_LD31621323.laz
2022-03-27 16:40:04   71207106 USGS_LPC_CO_EasternColorado_2018_A18_LD31621326.laz
2022-03-27 16:40:04   76174454 USGS_LPC_CO_EasternColorado_2018_A18_LD31621329.laz
2022-03-27 16:40:04   72297504 USGS_LPC_CO_EasternColorado_2018_A18_LD31621332.laz
2022-03-27 16:40:04   63454798 USGS_LPC_CO_EasternColorado_2018_A18_LD31621335.laz
2022-03-27 16:40:05   67549264 USGS_LPC_CO_EasternColorado_2018_A18_LD31621338.laz
2022-03-27 16:40:05   66453381 USGS_LPC_CO_EasternColorado_2018_A18_LD31621341.laz
2022-03-27 16:40:06   55144503 USGS_LPC_CO_EasternColorado_2018_A18_LD31621344.laz
2022-03-27 16:40:06   59936840 USGS_LPC_CO_EasternColorado_2018_A18_LD31621347.laz
2022

2022-03-27 16:41:19   75087644 USGS_LPC_CO_EasternColorado_2018_A18_LD31741329.laz
2022-03-27 16:41:20   75227076 USGS_LPC_CO_EasternColorado_2018_A18_LD31741332.laz
2022-03-27 16:41:20   67504688 USGS_LPC_CO_EasternColorado_2018_A18_LD31741335.laz
2022-03-27 16:41:21   66381670 USGS_LPC_CO_EasternColorado_2018_A18_LD31741338.laz
2022-03-27 16:41:21   61914172 USGS_LPC_CO_EasternColorado_2018_A18_LD31741341.laz
2022-03-27 16:41:21   59622291 USGS_LPC_CO_EasternColorado_2018_A18_LD31741344.laz
2022-03-27 16:41:22   59240890 USGS_LPC_CO_EasternColorado_2018_A18_LD31741347.laz
2022-03-27 16:41:22   58163905 USGS_LPC_CO_EasternColorado_2018_A18_LD31741350.laz
2022-03-27 16:41:22   64561317 USGS_LPC_CO_EasternColorado_2018_A18_LD31741353.laz
2022-03-27 16:41:22   58015691 USGS_LPC_CO_EasternColorado_2018_A18_LD31741356.laz
2022-03-27 16:41:22   51692704 USGS_LPC_CO_EasternColorado_2018_A18_LD31741359.laz
2022-03-27 16:41:23   50506194 USGS_LPC_CO_EasternColorado_2018_A18_LD31741362.laz
2022

2022-03-27 16:42:00   38767797 USGS_LPC_CO_EasternColorado_2018_A18_LD31801392.laz
2022-03-27 16:42:00   44053710 USGS_LPC_CO_EasternColorado_2018_A18_LD31801395.laz
2022-03-27 16:42:00   41404606 USGS_LPC_CO_EasternColorado_2018_A18_LD31801398.laz
2022-03-27 16:42:00   45418948 USGS_LPC_CO_EasternColorado_2018_A18_LD31801401.laz
2022-03-27 16:42:01   43517494 USGS_LPC_CO_EasternColorado_2018_A18_LD31801404.laz
2022-03-27 16:42:01   45436653 USGS_LPC_CO_EasternColorado_2018_A18_LD31801407.laz
2022-03-27 16:42:01   48920201 USGS_LPC_CO_EasternColorado_2018_A18_LD31801410.laz
2022-03-27 16:42:01   44144671 USGS_LPC_CO_EasternColorado_2018_A18_LD31801413.laz
2022-03-27 16:42:02   44736395 USGS_LPC_CO_EasternColorado_2018_A18_LD31801416.laz
2022-03-27 16:42:02   49995541 USGS_LPC_CO_EasternColorado_2018_A18_LD31801419.laz
2022-03-27 16:42:02   48263696 USGS_LPC_CO_EasternColorado_2018_A18_LD31801422.laz
2022-03-27 16:42:02   50229751 USGS_LPC_CO_EasternColorado_2018_A18_LD31801425.laz
2022

2022-03-27 16:42:50   47999770 USGS_LPC_CO_EasternColorado_2018_A18_LD31891371.laz
2022-03-27 16:42:50   45166189 USGS_LPC_CO_EasternColorado_2018_A18_LD31891374.laz
2022-03-27 16:42:50   48560999 USGS_LPC_CO_EasternColorado_2018_A18_LD31891377.laz
2022-03-27 16:42:50   46469656 USGS_LPC_CO_EasternColorado_2018_A18_LD31891380.laz
2022-03-27 16:42:51   44322764 USGS_LPC_CO_EasternColorado_2018_A18_LD31891383.laz
2022-03-27 16:42:51   43065265 USGS_LPC_CO_EasternColorado_2018_A18_LD31891386.laz
2022-03-27 16:42:51   48555780 USGS_LPC_CO_EasternColorado_2018_A18_LD31891389.laz
2022-03-27 16:42:51   49187538 USGS_LPC_CO_EasternColorado_2018_A18_LD31891392.laz
2022-03-27 16:42:52   46084910 USGS_LPC_CO_EasternColorado_2018_A18_LD31891395.laz
2022-03-27 16:42:52   50044590 USGS_LPC_CO_EasternColorado_2018_A18_LD31891398.laz
2022-03-27 16:42:52   52604825 USGS_LPC_CO_EasternColorado_2018_A18_LD31891401.laz
2022-03-27 16:42:52   56193337 USGS_LPC_CO_EasternColorado_2018_A18_LD31891404.laz
2022

2022-03-27 16:43:52   37544137 USGS_LPC_CO_EasternColorado_2018_A18_LD32011401.laz
2022-03-27 16:43:52   39756546 USGS_LPC_CO_EasternColorado_2018_A18_LD32011404.laz
2022-03-27 16:43:52   35841128 USGS_LPC_CO_EasternColorado_2018_A18_LD32011407.laz
2022-03-27 16:43:53   38163025 USGS_LPC_CO_EasternColorado_2018_A18_LD32011410.laz
2022-03-27 16:43:53  148487092 USGS_LPC_CO_EasternColorado_2018_A18_LD32011413.laz
2022-03-27 16:43:53   39348045 USGS_LPC_CO_EasternColorado_2018_A18_LD32011416.laz
2022-03-27 16:43:53   32457478 USGS_LPC_CO_EasternColorado_2018_A18_LD32011419.laz
2022-03-27 16:43:53   32318486 USGS_LPC_CO_EasternColorado_2018_A18_LD32011422.laz
2022-03-27 16:43:53   34982338 USGS_LPC_CO_EasternColorado_2018_A18_LD32011425.laz
2022-03-27 16:43:54   33517783 USGS_LPC_CO_EasternColorado_2018_A18_LD32011428.laz
2022-03-27 16:43:54   34140459 USGS_LPC_CO_EasternColorado_2018_A18_LD32011431.laz
2022-03-27 16:43:54   36869410 USGS_LPC_CO_EasternColorado_2018_A18_LD32011434.laz
2022

2022-03-27 16:45:02   54709097 USGS_LPC_CO_EasternColorado_2018_A18_LD32131368.laz
2022-03-27 16:45:02   56131136 USGS_LPC_CO_EasternColorado_2018_A18_LD32131371.laz
2022-03-27 16:45:02   55640701 USGS_LPC_CO_EasternColorado_2018_A18_LD32131374.laz
2022-03-27 16:45:02   58675732 USGS_LPC_CO_EasternColorado_2018_A18_LD32131377.laz
2022-03-27 16:45:03   54769590 USGS_LPC_CO_EasternColorado_2018_A18_LD32131380.laz
2022-03-27 16:45:03   50845765 USGS_LPC_CO_EasternColorado_2018_A18_LD32131383.laz
2022-03-27 16:45:04   48660954 USGS_LPC_CO_EasternColorado_2018_A18_LD32131386.laz
2022-03-27 16:45:04   47594554 USGS_LPC_CO_EasternColorado_2018_A18_LD32131389.laz
2022-03-27 16:45:04   50299705 USGS_LPC_CO_EasternColorado_2018_A18_LD32131392.laz
2022-03-27 16:45:05   50652535 USGS_LPC_CO_EasternColorado_2018_A18_LD32131395.laz
2022-03-27 16:45:05   44075598 USGS_LPC_CO_EasternColorado_2018_A18_LD32131398.laz
2022-03-27 16:45:05   44138880 USGS_LPC_CO_EasternColorado_2018_A18_LD32131401.laz
2022

2022-03-27 16:45:38   62136079 USGS_LPC_CO_EasternColorado_2018_A18_LD32191428.laz
2022-03-27 16:45:38   43236791 USGS_LPC_CO_EasternColorado_2018_A18_LD32191431.laz
2022-03-27 16:45:39   42247928 USGS_LPC_CO_EasternColorado_2018_A18_LD32191434.laz
2022-03-27 16:45:39   52828626 USGS_LPC_CO_EasternColorado_2018_A18_LD32191437.laz
2022-03-27 16:45:39   48833183 USGS_LPC_CO_EasternColorado_2018_A18_LD32191440.laz
2022-03-27 16:45:39   41781166 USGS_LPC_CO_EasternColorado_2018_A18_LD32191443.laz
2022-03-27 16:45:39   36412387 USGS_LPC_CO_EasternColorado_2018_A18_LD32191446.laz
2022-03-27 16:45:40   36304119 USGS_LPC_CO_EasternColorado_2018_A18_LD32191449.laz
2022-03-27 16:45:40   36469904 USGS_LPC_CO_EasternColorado_2018_A18_LD32191452.laz
2022-03-27 16:45:40   37744407 USGS_LPC_CO_EasternColorado_2018_A18_LD32191455.laz
2022-03-27 16:45:40   36073930 USGS_LPC_CO_EasternColorado_2018_A18_LD32191458.laz
2022-03-27 16:45:40   35738813 USGS_LPC_CO_EasternColorado_2018_A18_LD32191461.laz
2022

2022-03-27 16:45:58   37593507 USGS_LPC_CO_EasternColorado_2018_A18_LD32251299.laz
2022-03-27 16:45:59   36289734 USGS_LPC_CO_EasternColorado_2018_A18_LD32251302.laz
2022-03-27 16:45:59   38586735 USGS_LPC_CO_EasternColorado_2018_A18_LD32251305.laz
2022-03-27 16:45:59   42560561 USGS_LPC_CO_EasternColorado_2018_A18_LD32251308.laz
2022-03-27 16:45:59   42518151 USGS_LPC_CO_EasternColorado_2018_A18_LD32251311.laz
2022-03-27 16:45:59   38310484 USGS_LPC_CO_EasternColorado_2018_A18_LD32251314.laz
2022-03-27 16:46:00   37412022 USGS_LPC_CO_EasternColorado_2018_A18_LD32251317.laz
2022-03-27 16:46:00   37409603 USGS_LPC_CO_EasternColorado_2018_A18_LD32251320.laz
2022-03-27 16:46:00   35337200 USGS_LPC_CO_EasternColorado_2018_A18_LD32251323.laz
2022-03-27 16:46:00   41381367 USGS_LPC_CO_EasternColorado_2018_A18_LD32251326.laz
2022-03-27 16:46:00   36188094 USGS_LPC_CO_EasternColorado_2018_A18_LD32251329.laz
2022-03-27 16:46:00   31714733 USGS_LPC_CO_EasternColorado_2018_A18_LD32251332.laz
2022

2022-03-27 16:46:59   28409772 USGS_LPC_CO_EasternColorado_2018_A18_LD32371317.laz
2022-03-27 16:46:59   34500196 USGS_LPC_CO_EasternColorado_2018_A18_LD32371320.laz
2022-03-27 16:46:59   23788733 USGS_LPC_CO_EasternColorado_2018_A18_LD32371323.laz
2022-03-27 16:46:59   29510066 USGS_LPC_CO_EasternColorado_2018_A18_LD32371326.laz
2022-03-27 16:46:59   28641566 USGS_LPC_CO_EasternColorado_2018_A18_LD32371329.laz
2022-03-27 16:47:00   28477293 USGS_LPC_CO_EasternColorado_2018_A18_LD32371332.laz
2022-03-27 16:47:00   28939854 USGS_LPC_CO_EasternColorado_2018_A18_LD32371335.laz
2022-03-27 16:47:00   26036589 USGS_LPC_CO_EasternColorado_2018_A18_LD32371338.laz
2022-03-27 16:47:00   32391191 USGS_LPC_CO_EasternColorado_2018_A18_LD32371341.laz
2022-03-27 16:47:00   22851107 USGS_LPC_CO_EasternColorado_2018_A18_LD32371344.laz
2022-03-27 16:47:01   26696137 USGS_LPC_CO_EasternColorado_2018_A18_LD32371347.laz
2022-03-27 16:47:01   26060304 USGS_LPC_CO_EasternColorado_2018_A18_LD32371350.laz
2022

2022-03-27 16:48:03   28126932 USGS_LPC_CO_EasternColorado_2018_A18_LD32491350.laz
2022-03-27 16:48:03   27694306 USGS_LPC_CO_EasternColorado_2018_A18_LD32491353.laz
2022-03-27 16:48:03   28663405 USGS_LPC_CO_EasternColorado_2018_A18_LD32491356.laz
2022-03-27 16:48:03   29346116 USGS_LPC_CO_EasternColorado_2018_A18_LD32491359.laz
2022-03-27 16:48:03   30601440 USGS_LPC_CO_EasternColorado_2018_A18_LD32491362.laz
2022-03-27 16:48:03   29065890 USGS_LPC_CO_EasternColorado_2018_A18_LD32491365.laz
2022-03-27 16:48:03   31086824 USGS_LPC_CO_EasternColorado_2018_A18_LD32491368.laz
2022-03-27 16:48:04   30004694 USGS_LPC_CO_EasternColorado_2018_A18_LD32491371.laz
2022-03-27 16:48:04   32022671 USGS_LPC_CO_EasternColorado_2018_A18_LD32491374.laz
2022-03-27 16:48:04   33795998 USGS_LPC_CO_EasternColorado_2018_A18_LD32491377.laz
2022-03-27 16:48:04   33181649 USGS_LPC_CO_EasternColorado_2018_A18_LD32491380.laz
2022-03-27 16:48:04   32818979 USGS_LPC_CO_EasternColorado_2018_A18_LD32491383.laz
2022

2022-03-27 16:48:54   26647585 USGS_LPC_CO_EasternColorado_2018_A18_LD32611257.laz
2022-03-27 16:48:54   24822472 USGS_LPC_CO_EasternColorado_2018_A18_LD32611260.laz
2022-03-27 16:48:54   25751182 USGS_LPC_CO_EasternColorado_2018_A18_LD32611263.laz
2022-03-27 16:48:54   25733297 USGS_LPC_CO_EasternColorado_2018_A18_LD32611266.laz
2022-03-27 16:48:54   26673088 USGS_LPC_CO_EasternColorado_2018_A18_LD32611269.laz
2022-03-27 16:48:55   28368843 USGS_LPC_CO_EasternColorado_2018_A18_LD32611272.laz
2022-03-27 16:48:55   24306966 USGS_LPC_CO_EasternColorado_2018_A18_LD32611275.laz
2022-03-27 16:48:55   26245915 USGS_LPC_CO_EasternColorado_2018_A18_LD32611278.laz
2022-03-27 16:48:55   26806710 USGS_LPC_CO_EasternColorado_2018_A18_LD32611281.laz
2022-03-27 16:48:55   25816124 USGS_LPC_CO_EasternColorado_2018_A18_LD32611284.laz
2022-03-27 16:48:56   29148440 USGS_LPC_CO_EasternColorado_2018_A18_LD32611287.laz
2022-03-27 16:48:56   28942414 USGS_LPC_CO_EasternColorado_2018_A18_LD32611290.laz
2022

2022-03-27 16:49:20   37425619 USGS_LPC_CO_EasternColorado_2018_A18_LD32641458.laz
2022-03-27 16:49:20   36340810 USGS_LPC_CO_EasternColorado_2018_A18_LD32641461.laz
2022-03-27 16:49:20   34645376 USGS_LPC_CO_EasternColorado_2018_A18_LD32641464.laz
2022-03-27 16:49:21   38227261 USGS_LPC_CO_EasternColorado_2018_A18_LD32641467.laz
2022-03-27 16:49:21   39079408 USGS_LPC_CO_EasternColorado_2018_A18_LD32641470.laz
2022-03-27 16:49:21   45858945 USGS_LPC_CO_EasternColorado_2018_A18_LD32641473.laz
2022-03-27 16:49:21   24345662 USGS_LPC_CO_EasternColorado_2018_A18_LD32671251.laz
2022-03-27 16:49:21   35395337 USGS_LPC_CO_EasternColorado_2018_A18_LD32671254.laz
2022-03-27 16:49:22   34204969 USGS_LPC_CO_EasternColorado_2018_A18_LD32671257.laz
2022-03-27 16:49:22   36066343 USGS_LPC_CO_EasternColorado_2018_A18_LD32671260.laz
2022-03-27 16:49:22   36835125 USGS_LPC_CO_EasternColorado_2018_A18_LD32671263.laz
2022-03-27 16:49:22   35927253 USGS_LPC_CO_EasternColorado_2018_A18_LD32671266.laz
2022

2022-03-27 16:50:20   36212869 USGS_LPC_CO_EasternColorado_2018_A18_LD32761437.laz
2022-03-27 16:50:21   36466225 USGS_LPC_CO_EasternColorado_2018_A18_LD32761440.laz
2022-03-27 16:50:21   35374644 USGS_LPC_CO_EasternColorado_2018_A18_LD32761443.laz
2022-03-27 16:50:21   36446228 USGS_LPC_CO_EasternColorado_2018_A18_LD32761446.laz
2022-03-27 16:50:21   36199228 USGS_LPC_CO_EasternColorado_2018_A18_LD32761449.laz
2022-03-27 16:50:21   43981013 USGS_LPC_CO_EasternColorado_2018_A18_LD32761452.laz
2022-03-27 16:50:22   34835628 USGS_LPC_CO_EasternColorado_2018_A18_LD32761455.laz
2022-03-27 16:50:22   36016444 USGS_LPC_CO_EasternColorado_2018_A18_LD32761458.laz
2022-03-27 16:50:22   39094634 USGS_LPC_CO_EasternColorado_2018_A18_LD32761461.laz
2022-03-27 16:50:22   43015534 USGS_LPC_CO_EasternColorado_2018_A18_LD32761464.laz
2022-03-27 16:50:23   48974971 USGS_LPC_CO_EasternColorado_2018_A18_LD32761467.laz
2022-03-27 16:50:23   43466326 USGS_LPC_CO_EasternColorado_2018_A18_LD32761470.laz
2022

2022-03-27 16:51:14   35594367 USGS_LPC_CO_EasternColorado_2018_A18_LD32881419.laz
2022-03-27 16:51:14   34760483 USGS_LPC_CO_EasternColorado_2018_A18_LD32881422.laz
2022-03-27 16:51:14   34631039 USGS_LPC_CO_EasternColorado_2018_A18_LD32881425.laz
2022-03-27 16:51:15   36550569 USGS_LPC_CO_EasternColorado_2018_A18_LD32881428.laz
2022-03-27 16:51:15   35486489 USGS_LPC_CO_EasternColorado_2018_A18_LD32881431.laz
2022-03-27 16:51:15   35994648 USGS_LPC_CO_EasternColorado_2018_A18_LD32881434.laz
2022-03-27 16:51:15   36728213 USGS_LPC_CO_EasternColorado_2018_A18_LD32881437.laz
2022-03-27 16:51:15   35105158 USGS_LPC_CO_EasternColorado_2018_A18_LD32881440.laz
2022-03-27 16:51:15   35192050 USGS_LPC_CO_EasternColorado_2018_A18_LD32881443.laz
2022-03-27 16:51:16   35126633 USGS_LPC_CO_EasternColorado_2018_A18_LD32881446.laz
2022-03-27 16:51:16   34374957 USGS_LPC_CO_EasternColorado_2018_A18_LD32881449.laz
2022-03-27 16:51:16   37472672 USGS_LPC_CO_EasternColorado_2018_A18_LD32881452.laz
2022

2022-03-27 16:52:06   36021910 USGS_LPC_CO_EasternColorado_2018_A18_LD33001332.laz
2022-03-27 16:52:06   37462580 USGS_LPC_CO_EasternColorado_2018_A18_LD33001335.laz
2022-03-27 16:52:06   36530686 USGS_LPC_CO_EasternColorado_2018_A18_LD33001338.laz
2022-03-27 16:52:07   37483256 USGS_LPC_CO_EasternColorado_2018_A18_LD33001341.laz
2022-03-27 16:52:07   35378456 USGS_LPC_CO_EasternColorado_2018_A18_LD33001344.laz
2022-03-27 16:52:07   38699994 USGS_LPC_CO_EasternColorado_2018_A18_LD33001347.laz
2022-03-27 16:52:08   36385256 USGS_LPC_CO_EasternColorado_2018_A18_LD33001350.laz
2022-03-27 16:52:08   34230841 USGS_LPC_CO_EasternColorado_2018_A18_LD33001353.laz
2022-03-27 16:52:08   36636858 USGS_LPC_CO_EasternColorado_2018_A18_LD33001356.laz
2022-03-27 16:52:08   36828424 USGS_LPC_CO_EasternColorado_2018_A18_LD33001359.laz
2022-03-27 16:52:08   33591583 USGS_LPC_CO_EasternColorado_2018_A18_LD33001362.laz
2022-03-27 16:52:09   37236680 USGS_LPC_CO_EasternColorado_2018_A18_LD33001365.laz
2022

2022-03-27 16:52:36   28857045 USGS_LPC_CO_EasternColorado_2018_A18_LD33061332.laz
2022-03-27 16:52:36   28727898 USGS_LPC_CO_EasternColorado_2018_A18_LD33061335.laz
2022-03-27 16:52:36   27771271 USGS_LPC_CO_EasternColorado_2018_A18_LD33061338.laz
2022-03-27 16:52:36   27529674 USGS_LPC_CO_EasternColorado_2018_A18_LD33061341.laz
2022-03-27 16:52:36   27019487 USGS_LPC_CO_EasternColorado_2018_A18_LD33061344.laz
2022-03-27 16:52:36   28101921 USGS_LPC_CO_EasternColorado_2018_A18_LD33061347.laz
2022-03-27 16:52:37   29011385 USGS_LPC_CO_EasternColorado_2018_A18_LD33061350.laz
2022-03-27 16:52:37   30077062 USGS_LPC_CO_EasternColorado_2018_A18_LD33061353.laz
2022-03-27 16:52:37   26711633 USGS_LPC_CO_EasternColorado_2018_A18_LD33061356.laz
2022-03-27 16:52:37   29339499 USGS_LPC_CO_EasternColorado_2018_A18_LD33061359.laz
2022-03-27 16:52:37   25595473 USGS_LPC_CO_EasternColorado_2018_A18_LD33061362.laz
2022-03-27 16:52:38   29696923 USGS_LPC_CO_EasternColorado_2018_A18_LD33061365.laz
2022

2022-03-27 16:53:31   30338751 USGS_LPC_CO_EasternColorado_2018_A18_LD33181353.laz
2022-03-27 16:53:31   30112098 USGS_LPC_CO_EasternColorado_2018_A18_LD33181356.laz
2022-03-27 16:53:31   31446228 USGS_LPC_CO_EasternColorado_2018_A18_LD33181359.laz
2022-03-27 16:53:32   30758546 USGS_LPC_CO_EasternColorado_2018_A18_LD33181362.laz
2022-03-27 16:53:32   31184474 USGS_LPC_CO_EasternColorado_2018_A18_LD33181365.laz
2022-03-27 16:53:32   33071000 USGS_LPC_CO_EasternColorado_2018_A18_LD33181368.laz
2022-03-27 16:53:32   33953286 USGS_LPC_CO_EasternColorado_2018_A18_LD33181371.laz
2022-03-27 16:53:32   32808892 USGS_LPC_CO_EasternColorado_2018_A18_LD33181374.laz
2022-03-27 16:53:32   34065466 USGS_LPC_CO_EasternColorado_2018_A18_LD33181377.laz
2022-03-27 16:53:32   35696112 USGS_LPC_CO_EasternColorado_2018_A18_LD33181380.laz
2022-03-27 16:53:33   35930563 USGS_LPC_CO_EasternColorado_2018_A18_LD33181383.laz
2022-03-27 16:53:33   37591050 USGS_LPC_CO_EasternColorado_2018_A18_LD33181386.laz
2022

2022-03-27 16:54:27   37218109 USGS_LPC_CO_EasternColorado_2018_A18_LD33301392.laz
2022-03-27 16:54:27   36575340 USGS_LPC_CO_EasternColorado_2018_A18_LD33301395.laz
2022-03-27 16:54:27   37183107 USGS_LPC_CO_EasternColorado_2018_A18_LD33301398.laz
2022-03-27 16:54:27   35962741 USGS_LPC_CO_EasternColorado_2018_A18_LD33301401.laz
2022-03-27 16:54:28   36101534 USGS_LPC_CO_EasternColorado_2018_A18_LD33301404.laz
2022-03-27 16:54:28   36617000 USGS_LPC_CO_EasternColorado_2018_A18_LD33301407.laz
2022-03-27 16:54:28   37548337 USGS_LPC_CO_EasternColorado_2018_A18_LD33301410.laz
2022-03-27 16:54:28   37945070 USGS_LPC_CO_EasternColorado_2018_A18_LD33301413.laz
2022-03-27 16:54:28   36854093 USGS_LPC_CO_EasternColorado_2018_A18_LD33301416.laz
2022-03-27 16:54:29   35748433 USGS_LPC_CO_EasternColorado_2018_A18_LD33301419.laz
2022-03-27 16:54:29   36200501 USGS_LPC_CO_EasternColorado_2018_A18_LD33301422.laz
2022-03-27 16:54:29   36507650 USGS_LPC_CO_EasternColorado_2018_A18_LD33301425.laz
2022

2022-03-27 16:55:11   33658362 USGS_LPC_CO_EasternColorado_2018_A18_LD33391407.laz
2022-03-27 16:55:12   31484637 USGS_LPC_CO_EasternColorado_2018_A18_LD33391410.laz
2022-03-27 16:55:12   32586680 USGS_LPC_CO_EasternColorado_2018_A18_LD33391413.laz
2022-03-27 16:55:12   30480270 USGS_LPC_CO_EasternColorado_2018_A18_LD33391416.laz
2022-03-27 16:55:12   30857171 USGS_LPC_CO_EasternColorado_2018_A18_LD33391419.laz
2022-03-27 16:55:12   30433929 USGS_LPC_CO_EasternColorado_2018_A18_LD33391422.laz
2022-03-27 16:55:13   30975087 USGS_LPC_CO_EasternColorado_2018_A18_LD33391425.laz
2022-03-27 16:55:13   29667643 USGS_LPC_CO_EasternColorado_2018_A18_LD33391428.laz
2022-03-27 16:55:13   28878917 USGS_LPC_CO_EasternColorado_2018_A18_LD33391431.laz
2022-03-27 16:55:13   28798909 USGS_LPC_CO_EasternColorado_2018_A18_LD33391434.laz
2022-03-27 16:55:13   31326584 USGS_LPC_CO_EasternColorado_2018_A18_LD33391437.laz
2022-03-27 16:55:13   30550052 USGS_LPC_CO_EasternColorado_2018_A18_LD33391440.laz
2022

2022-03-27 16:56:20   31772500 USGS_LPC_CO_EasternColorado_2018_A18_LD33541428.laz
2022-03-27 16:56:20   30899058 USGS_LPC_CO_EasternColorado_2018_A18_LD33541431.laz
2022-03-27 16:56:21   31563084 USGS_LPC_CO_EasternColorado_2018_A18_LD33541434.laz
2022-03-27 16:56:21   31697730 USGS_LPC_CO_EasternColorado_2018_A18_LD33541437.laz
2022-03-27 16:56:21   31193041 USGS_LPC_CO_EasternColorado_2018_A18_LD33541440.laz
2022-03-27 16:56:21   28982594 USGS_LPC_CO_EasternColorado_2018_A18_LD33541443.laz
2022-03-27 16:56:21   30396992 USGS_LPC_CO_EasternColorado_2018_A18_LD33541446.laz
2022-03-27 16:56:21   31716945 USGS_LPC_CO_EasternColorado_2018_A18_LD33541449.laz
2022-03-27 16:56:21   31043873 USGS_LPC_CO_EasternColorado_2018_A18_LD33541452.laz
2022-03-27 16:56:22   32178672 USGS_LPC_CO_EasternColorado_2018_A18_LD33541455.laz
2022-03-27 16:56:22   29089425 USGS_LPC_CO_EasternColorado_2018_A18_LD33541458.laz
2022-03-27 16:56:22   28354241 USGS_LPC_CO_EasternColorado_2018_A18_LD33541461.laz
2022

2022-03-27 16:57:17   29456622 USGS_LPC_CO_EasternColorado_2018_A18_LD33661458.laz
2022-03-27 16:57:17   29601472 USGS_LPC_CO_EasternColorado_2018_A18_LD33661461.laz
2022-03-27 16:57:18   30047083 USGS_LPC_CO_EasternColorado_2018_A18_LD33661464.laz
2022-03-27 16:57:18   28319658 USGS_LPC_CO_EasternColorado_2018_A18_LD33661467.laz
2022-03-27 16:57:18   31297832 USGS_LPC_CO_EasternColorado_2018_A18_LD33661470.laz
2022-03-27 16:57:18   34103096 USGS_LPC_CO_EasternColorado_2018_A18_LD33661473.laz
2022-03-27 16:57:18   19865210 USGS_LPC_CO_EasternColorado_2018_A18_LD33691251.laz
2022-03-27 16:57:18   56315429 USGS_LPC_CO_EasternColorado_2018_A18_LD33691254.laz
2022-03-27 16:57:19   30877666 USGS_LPC_CO_EasternColorado_2018_A18_LD33691257.laz
2022-03-27 16:57:19   28335070 USGS_LPC_CO_EasternColorado_2018_A18_LD33691260.laz
2022-03-27 16:57:19   28077441 USGS_LPC_CO_EasternColorado_2018_A18_LD33691263.laz
2022-03-27 16:57:19   27644274 USGS_LPC_CO_EasternColorado_2018_A18_LD33691266.laz
2022

2022-03-27 16:58:08   30316146 USGS_LPC_CO_EasternColorado_2018_A18_LD33781434.laz
2022-03-27 16:58:09   31047435 USGS_LPC_CO_EasternColorado_2018_A18_LD33781437.laz
2022-03-27 16:58:09   29783042 USGS_LPC_CO_EasternColorado_2018_A18_LD33781440.laz
2022-03-27 16:58:09   31669118 USGS_LPC_CO_EasternColorado_2018_A18_LD33781443.laz
2022-03-27 16:58:09   32771319 USGS_LPC_CO_EasternColorado_2018_A18_LD33781446.laz
2022-03-27 16:58:09   34730042 USGS_LPC_CO_EasternColorado_2018_A18_LD33781449.laz
2022-03-27 16:58:09   33926242 USGS_LPC_CO_EasternColorado_2018_A18_LD33781452.laz
2022-03-27 16:58:10   34170871 USGS_LPC_CO_EasternColorado_2018_A18_LD33781455.laz
2022-03-27 16:58:10   36236965 USGS_LPC_CO_EasternColorado_2018_A18_LD33781458.laz
2022-03-27 16:58:10   36727083 USGS_LPC_CO_EasternColorado_2018_A18_LD33781461.laz
2022-03-27 16:58:10   37231688 USGS_LPC_CO_EasternColorado_2018_A18_LD33781464.laz
2022-03-27 16:58:10   36872428 USGS_LPC_CO_EasternColorado_2018_A18_LD33781467.laz
2022

2022-03-27 16:58:30   36042567 USGS_LPC_CO_EasternColorado_2018_A18_LD33841332.laz
2022-03-27 16:58:30   35597602 USGS_LPC_CO_EasternColorado_2018_A18_LD33841335.laz
2022-03-27 16:58:30   36352927 USGS_LPC_CO_EasternColorado_2018_A18_LD33841338.laz
2022-03-27 16:58:30   37549798 USGS_LPC_CO_EasternColorado_2018_A18_LD33841341.laz
2022-03-27 16:58:30   38846204 USGS_LPC_CO_EasternColorado_2018_A18_LD33841344.laz
2022-03-27 16:58:31   38513723 USGS_LPC_CO_EasternColorado_2018_A18_LD33841347.laz
2022-03-27 16:58:31   37616836 USGS_LPC_CO_EasternColorado_2018_A18_LD33841350.laz
2022-03-27 16:58:31   38334222 USGS_LPC_CO_EasternColorado_2018_A18_LD33841353.laz
2022-03-27 16:58:31   36077502 USGS_LPC_CO_EasternColorado_2018_A18_LD33841356.laz
2022-03-27 16:58:31   35712104 USGS_LPC_CO_EasternColorado_2018_A18_LD33841359.laz
2022-03-27 16:58:32   37236072 USGS_LPC_CO_EasternColorado_2018_A18_LD33841362.laz
2022-03-27 16:58:32   35981190 USGS_LPC_CO_EasternColorado_2018_A18_LD33841365.laz
2022

2022-03-27 16:59:20   32077517 USGS_LPC_CO_EasternColorado_2018_A18_LD33961290.laz
2022-03-27 16:59:21   31957140 USGS_LPC_CO_EasternColorado_2018_A18_LD33961293.laz
2022-03-27 16:59:20   33167120 USGS_LPC_CO_EasternColorado_2018_A18_LD33961296.laz
2022-03-27 16:59:21   33865140 USGS_LPC_CO_EasternColorado_2018_A18_LD33961299.laz
2022-03-27 16:59:21   34131505 USGS_LPC_CO_EasternColorado_2018_A18_LD33961302.laz
2022-03-27 16:59:21   36931348 USGS_LPC_CO_EasternColorado_2018_A18_LD33961305.laz
2022-03-27 16:59:21   33094209 USGS_LPC_CO_EasternColorado_2018_A18_LD33961308.laz
2022-03-27 16:59:21   36513077 USGS_LPC_CO_EasternColorado_2018_A18_LD33961311.laz
2022-03-27 16:59:22   32676649 USGS_LPC_CO_EasternColorado_2018_A18_LD33961314.laz
2022-03-27 16:59:22   36238187 USGS_LPC_CO_EasternColorado_2018_A18_LD33961317.laz
2022-03-27 16:59:22   32315209 USGS_LPC_CO_EasternColorado_2018_A18_LD33961320.laz
2022-03-27 16:59:22   35920559 USGS_LPC_CO_EasternColorado_2018_A18_LD33961323.laz
2022

2022-03-27 17:00:11   35376059 USGS_LPC_CO_EasternColorado_2018_A18_LD34051464.laz
2022-03-27 17:00:11   36427457 USGS_LPC_CO_EasternColorado_2018_A18_LD34051467.laz
2022-03-27 17:00:11   35514514 USGS_LPC_CO_EasternColorado_2018_A18_LD34051470.laz
2022-03-27 17:00:11   35842347 USGS_LPC_CO_EasternColorado_2018_A18_LD34051473.laz
2022-03-27 17:00:12    8899780 USGS_LPC_CO_EasternColorado_2018_A18_LD34081251.laz
2022-03-27 17:00:12   82526844 USGS_LPC_CO_EasternColorado_2018_A18_LD34081254.laz
2022-03-27 17:00:12   62766315 USGS_LPC_CO_EasternColorado_2018_A18_LD34081257.laz
2022-03-27 17:00:12   34175478 USGS_LPC_CO_EasternColorado_2018_A18_LD34081260.laz
2022-03-27 17:00:12   33744748 USGS_LPC_CO_EasternColorado_2018_A18_LD34081263.laz
2022-03-27 17:00:13   34046543 USGS_LPC_CO_EasternColorado_2018_A18_LD34081266.laz
2022-03-27 17:00:14   36289438 USGS_LPC_CO_EasternColorado_2018_A18_LD34081269.laz
2022-03-27 17:00:14   38400854 USGS_LPC_CO_EasternColorado_2018_A18_LD34081272.laz
2022

In [7]:
%%bash

aws s3 cp s3://eccoe-lidar/CO_Eastern_ElPaso_2018/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz ~

download: s3://eccoe-lidar/CO_Eastern_ElPaso_2018/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz to ../../../../../../../../home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz


In [8]:
! ls -l /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz

-rw-rw-r-- 1 ec2-user ec2-user 11449736 Mar 27 16:37 /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz


In [9]:
# pdal info ./exercises/info/interesting.las -p 0

In [11]:
! pdal info /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz -p 0

{
  "file_size": 11449736,
  "filename": "/home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz",
  "now": "2022-04-22T18:15:36+0000",
  "pdal_version": "2.4.0 (git-version: 997a1c)",
  "points":
  {
    "point":
    {
      "ClassFlags": 0,
      "Classification": 1,
      "EdgeOfFlightLine": 0,
      "GpsTime": 215001970.7,
      "Intensity": 1280,
      "NumberOfReturns": 3,
      "PointId": 0,
      "PointSourceId": 1283,
      "ReturnNumber": 1,
      "ScanAngleRank": -4.782000065,
      "ScanChannel": 0,
      "ScanDirectionFlag": 0,
      "UserData": 0,
      "X": 3121871.315,
      "Y": 1351901.555,
      "Z": 11429.921
    }
  },
  "reader": "readers.las"
}


In [12]:
! pdal info --all /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz 

{
  "boundary":
  {
    "area": 11577110.97,
    "avg_pt_per_sq_unit": 51.50138588,
    "avg_pt_spacing": 3.782319462,
    "boundary": "POLYGON ((3123206.38534872 1349589.14532376,3123873.92052308 1353057.75983812,3121871.315 1354213.96467624,3119201.17430256 1351901.555,3123206.38534872 1349589.14532376))",
    "boundary_json": { "type": "Polygon", "coordinates": [ [ [ 3123206.38534872001037, 1349589.145323760109022 ], [ 3123873.920523080043495, 1353057.759838120080531 ], [ 3121871.315, 1354213.964676239993423 ], [ 3119201.174302559811622, 1351901.555 ], [ 3123206.38534872001037, 1349589.145323760109022 ] ] ] },
    "density": 0.0699010316,
    "edge_length": 0,
    "estimated_edge": 2312.409676,
    "hex_offsets": "MULTIPOINT (0 0, -667.535 1156.2, 0 2312.41, 1335.07 2312.41, 2002.61 1156.2, 1335.07 0)",
    "sample_size": 5000,
    "threshold": 15
  },
  "file_size": 11449736,
  "filename": "/home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz",
  "metadata":
  {
    "

In [13]:
! pdal info --help

usage: pdal info [options] [input]
standard options:
  --developer-debug  Enable developer debug (don't trap exceptions)
  --label            A string to label the process with
  --driver           Override reader driver
  --help             Print help and exit
options:
  --input, -i               Input file name
  --all                     Dump statistics, schema and metadata
  --point, -p               Point to dump --point="1-5,10,100-200" (0 indexed)
  --query                   Return points in order of distance from the
      specified location (2D or 3D) --query Xcoord,Ycoord[,Zcoord][/count]
  --stats                   Dump stats on all points (reads entire dataset)
  --boundary                Compute a hexagonal hull/boundary of dataset
  --dimensions              Dimensions on which to compute statistics
  --enumerate               Dimensions whose values should be enumerated
  --schema                  Dump the schema
  --pipeline-serialization  Output filename for pipeline s

In [14]:
! pdal info --stats /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz 

{
  "file_size": 11449736,
  "filename": "/home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz",
  "now": "2022-04-22T18:24:45+0000",
  "pdal_version": "2.4.0 (git-version: 997a1c)",
  "reader": "readers.las",
  "stats":
  {
    "bbox":
    {
      "EPSG:4326":
      {
        "bbox":
        {
          "maxx": -105.0683947,
          "maxy": 38.80180991,
          "maxz": 3549.32779,
          "minx": -105.0735521,
          "miny": 38.79826988,
          "minz": 3370.954449
        },
        "boundary": { "type": "Polygon", "coordinates": [ [ [ -105.073552081963371, 38.798288864625327, 3370.954449180283973 ], [ -105.073530959501483, 38.801809906631853, 3370.968886481337449 ], [ -105.068394654301272, 38.801790925117309, 3549.327790134110273 ], [ -105.068416031150122, 38.798269884045524, 3549.313437282704399 ], [ -105.073552081963371, 38.798288864625327, 3370.954449180283973 ] ] ] }
      },
      "native":
      {
        "bbox":
        {
          "maxx": 3122999.993

In [15]:
! pdal info --all /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz | grep -i epsg


    "comp_spatialreference": "COMPD_CS[\"NAD83(2011) / Colorado Central (ftUS) + NAVD88 height (ftUS) - US Geoid Model of 2012 B\",\n    PROJCS[\"NAD83(2011) / Colorado Central (ftUS)\",\n        GEOGCS[\"NAD83(2011)\",\n            DATUM[\"NAD83_National_Spatial_Reference_System_2011\",\n                SPHEROID[\"GRS 1980\",6378137,298.257222101,\n                    AUTHORITY[\"EPSG\",\"7019\"]],\n                AUTHORITY[\"EPSG\",\"1116\"]],\n            PRIMEM[\"Greenwich\",0,\n                AUTHORITY[\"EPSG\",\"8901\"]],\n            UNIT[\"degree\",0.0174532925199433,\n                AUTHORITY[\"EPSG\",\"9122\"]],\n            AUTHORITY[\"EPSG\",\"6318\"]],\n        PROJECTION[\"Lambert_Conformal_Conic_2SP\"],\n        PARAMETER[\"standard_parallel_1\",39.75],\n        PARAMETER[\"standard_parallel_2\",38.45],\n        PARAMETER[\"latitude_of_origin\",37.83333333333334],\n        PARAMETER[\"central_meridian\",-105.5],\n        PARAMETER[\"false_easting\",3000000],\n        

# https://epsg.io/6428

- EPSG:6428
- NAD83(2011) / Colorado Central (ftUS)



# Exercise - use proj4 to verify 4326 bounding box from EPSG:6428

In [18]:
! jsawk --help

/bin/bash: jsawk: command not found


In [19]:
! jq --help

jq - commandline JSON processor [version 1.5]
Usage: jq [options] <jq filter> [file...]

	jq is a tool for processing JSON inputs, applying the
	given filter to its JSON text inputs and producing the
	filter's results as JSON on standard output.
	The simplest filter is ., which is the identity filter,
	copying jq's input to its output unmodified (except for
	formatting).
	For more advanced filters see the jq(1) manpage ("man jq")
	and/or https://stedolan.github.io/jq

	Some of the options include:
	 -c		compact instead of pretty-printed output;
	 -n		use `null` as the single input value;
	 -e		set the exit status code based on the output;
	 -s		read (slurp) all inputs into an array; apply filter to it;
	 -r		output raw strings, not JSON texts;
	 -R		read raw strings, not JSON texts;
	 -C		colorize JSON;
	 -M		monochrome (don't colorize JSON);
	 -S		sort keys of objects on output;
	 --tab	use tabs for indentation;
	 --arg a v	set variable $a to value <v>;
	 --argjson a v	set variable $a

In [23]:
! pdal info --metadata /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz | jq ".metadata.compressed, .metadata.system_id"


true
"LAStools (c) by rapidlasso GmbH"


In [25]:
!pdal info /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz --metadata

{
  "file_size": 11449736,
  "filename": "/home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz",
  "metadata":
  {
    "comp_spatialreference": "COMPD_CS[\"NAD83(2011) / Colorado Central (ftUS) + NAVD88 height (ftUS) - US Geoid Model of 2012 B\",\n    PROJCS[\"NAD83(2011) / Colorado Central (ftUS)\",\n        GEOGCS[\"NAD83(2011)\",\n            DATUM[\"NAD83_National_Spatial_Reference_System_2011\",\n                SPHEROID[\"GRS 1980\",6378137,298.257222101,\n                    AUTHORITY[\"EPSG\",\"7019\"]],\n                AUTHORITY[\"EPSG\",\"1116\"]],\n            PRIMEM[\"Greenwich\",0,\n                AUTHORITY[\"EPSG\",\"8901\"]],\n            UNIT[\"degree\",0.0174532925199433,\n                AUTHORITY[\"EPSG\",\"9122\"]],\n            AUTHORITY[\"EPSG\",\"6318\"]],\n        PROJECTION[\"Lambert_Conformal_Conic_2SP\"],\n        PARAMETER[\"standard_parallel_1\",39.75],\n        PARAMETER[\"standard_parallel_2\",38.45],\n        PARAMETER[\"latitude_of_origi

In [28]:
!pdal info /home/ec2-user/USGS_LPC_CO_EasternColorado_2018_A18_LD31201350.laz --metadata | jq ".metadata.srs.wkt"

"PROJCS[\"NAD83(2011) / Colorado Central (ftUS)\",GEOGCS[\"NAD83(2011)\",DATUM[\"NAD83_National_Spatial_Reference_System_2011\",SPHEROID[\"GRS 1980\",6378137,298.257222101,AUTHORITY[\"EPSG\",\"7019\"]],AUTHORITY[\"EPSG\",\"1116\"]],PRIMEM[\"Greenwich\",0,AUTHORITY[\"EPSG\",\"8901\"]],UNIT[\"degree\",0.0174532925199433,AUTHORITY[\"EPSG\",\"9122\"]],AUTHORITY[\"EPSG\",\"6318\"]],PROJECTION[\"Lambert_Conformal_Conic_2SP\"],PARAMETER[\"standard_parallel_1\",39.75],PARAMETER[\"standard_parallel_2\",38.45],PARAMETER[\"latitude_of_origin\",37.8333333333333],PARAMETER[\"central_meridian\",-105.5],PARAMETER[\"false_easting\",3000000],PARAMETER[\"false_northing\",1000000],UNIT[\"US survey foot\",0.304800609601219,AUTHORITY[\"EPSG\",\"9003\"]],AXIS[\"X\",EAST],AXIS[\"Y\",NORTH],AUTHORITY[\"EPSG\",\"6428\"]]"
